api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -202,6 +202,27 @@ message HostCertsRequest {
   string UnstableSystemRoleAssertionID = 12 [(gogoproto.jsontag) = "system_role_assertion_id,omitempty"];
 }
 
+// OpenSSHCertRequest specifies certificate-generation parameters
+// for a certificates used to connect to Agentless nodes.
+message OpenSSHCertRequest {
```

Comment:
I think it will also need `LoginIP` field - in `generateUserCert()` we check if user has IP pinning enabled (`req.checker.PinSourceIP()`) and if they have, we require login IP to be present so it could be embedded into the certificate. Right now it only issues warning if IP is missing, but the plan is to make it a hard error.

Reply:
I originally had that, but after discussing with @jakule we decided to remove it. The certificate returned by `GenerateOpenSSHCert` should only be used to authenticate with OpenSSH servers, which have no way to dynamically verify allowed IPs in a certificate. Apparently it would be possible to specify what IPs are allowed to connect in the `TrustedUserCAKeys` certificate, which for Agentless nodes is the OpenSSH CA.

When login IPs are strictly enforced, we can instead enforce this at the Proxy rather than trying to make it work with OpenSSH, as the Proxy is creating and forwarding a SSH connection (when connecting to Agentless nodes) so it knows the true client IP. Make sense?

Reply:
And connection will always come from the proxy itself?
Yep, makes sense, I'm just thinking how we can gracefully handle it in the `generateCert()` 🤔

Reply:
That would work, then maybe `reversetunnel.RemoteSite.Dial` could do the IP check if the target node is Agentless? That would avoid the callers of `proxy.Router.DialHost` having to preform the check themselves

Reply:
Well maybe `proxy.Router.DialHost` could do the check then, otherwise the check would have to be done in multiple places IIRC. Regardless, shouldn't affect this PR right?

Reply:
@AntonAM I want to make sure that we're on the same page. `OpenSSH CA` is somehow unique. This certificate is only used when we call `OpenSSH` server, so adding any fields to it doesn't make any difference as `sshd` is not able to verify anything.

I agree that the proxy could do the check, but (correct me if I'm wrong) isn't it the case that the IP is already a part of the user certificate and it's checked when you connect to the proxy? In this case, this check should be already implemented.

Reply:
@jakule yep, makes sense for `OpenSSH CA`, we can make a check in `generateCertificate` and don't enforce IP pinning on it, it's ok as long as user doesn't have any way to get that intermediate certificate.
We rely on go `ssh` package to enforce IP pinning for SSH connection, so as long as we terminate SSH in proxy, it should be checked. As I see here we always go through intermediary in-memory SSH server in the proxy for OpenSSH connection, so in that case yep, we're good, existing IP pinning infrastructure should just work 👍 

@capnspacehook yep, on second look, I think we're good and no additional checks required 👍 :)